x-pack/plugins/viz_editor/pie_chart_plugin/pie_chart_plugin.tsx

```diff
@@ -129,11 +132,71 @@ function getSuggestion(visModel: PieChartVisModel) {
   };
 }
 
+function getSuggestionsForField(datasourceRef: string, field: Field, visModel: PieChartVisModel) {
+  const operationNames = getOperationsForField(field);
+
+  if (operationNames.length === 0) {
+    return [];
+  }
+
+  let firstOperation: SelectOperation;
+
+  // Remapping for pie chart to make sense
+  if (operationNames[0] === 'terms') {
+    firstOperation = {
+      operation: 'column',
+      argument: { field: field.name, size: 5 },
+    };
+  } else if (operationNames[0] === 'date_histogram') {
+    firstOperation = {
+      operation: 'date_histogram',
+      argument: { field: field.name, interval: '1d' },
+    };
+  } else if (operationNames[0] === 'count') {
+    firstOperation = { operation: 'column', argument: { field: field.name } };
+  } else {
+    firstOperation = {
+      operation: operationNames[0],
```

Comment:
This is a bit of an assumption about what a default should be for any operations not covered. I think I'd prefer if we had a `generateDefault` or something that gets added to `selectOperations` in `/server/query/select_operations.ts` and maybe that file (or most of the server query, folder) gets moved up into common. That way, if we forget to define a default for an operation, we get a nice error rather than an unexpected behavior.

Reply:
I will be splitting the logic around operations into a separate PR because it deserves to be more complete.